src/playbacks/hls/hls.js

```diff
@@ -119,6 +134,25 @@ export default class HLS extends HTML5VideoPlayback {
     this.trigger(Events.PLAYBACK_SETTINGSUPDATE)
   }
 
+  onError(evt, data) {
+    if (data && data.fatal && this.recoverAttemptsRemaining > 0) {
+      this.recoverAttemptsRemaining -= 1
+      switch (data.type) {
+        case HLSJS.ErrorTypes.NETWORK_ERROR:
+          Log.warning(`hlsjs: trying to recover from network error, evt ${evt}, data ${data} `)
+          this.hls.startLoad()
+          break
+        case HLSJS.ErrorTypes.MEDIA_ERROR:
+          Log.warning(`hlsjs: trying to recover from media error, evt ${evt}, data ${data} `)
+          this.recover()
+          break
+        default:
+          Log.error(`hlsjs: could not recover from error, evt ${evt}, data ${data} `)
+          break
+      }
```

Comment:
You're probably right, we need to trigger PLAYBACK_ERROR for default and the failed recover, good catch.